src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WidgetsCombinedTests.java

```diff
@@ -0,0 +1,102 @@
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @ownership: Content X-Wing
+ */
+public class WidgetsCombinedTests extends NewTestTemplate {
+
+  private static final String ARTICLE_NAME = "WidgetsCombined";
+  private static final String MAPS_ARTICLE_NAME = "Map";
+  private static ArrayList<WidgetPageObject> widgets = new ArrayList<WidgetPageObject>();
+
+  @BeforeMethod(alwaysRun = true)
+  public void prepareTest() {
+    driver.manage().timeouts().pageLoadTimeout(100, TimeUnit.SECONDS);
+    widgets.add(new PollsnackWidgetPageObject(driver));
+    widgets.add(new SoundCloudWidgetPageObject(driver));
+    widgets.add(new SpotifyWidgetPageObject(driver));
+    widgets.add(new TwitterWidgetPageObject(driver));
+    widgets.add(new VKWidgetPageObject(driver));
+    widgets.add(new WeiboWidgetPageObject(driver));
+  }
+
+  @Test(groups = "MercuryWidgetsCombinedTest_001")
+  @Execute(onWikia = "mercuryautomationtesting")
+  public void MercuryWidgetsCombinedTest_001_isLoadedOnFirstVisitDirectlyFromUrl() {
+    String content = "";
+    for (WidgetPageObject widget : widgets) {
+      content += widget.getTag();
+    }
+
+    ArticleContent articleContent = new ArticleContent();
+    articleContent.clear(ARTICLE_NAME);
+    articleContent.push(content, ARTICLE_NAME);
+
+    ArticlePageObject article = new ArticlePageObject(driver);
+    article.open(ARTICLE_NAME);
+
+    for (WidgetPageObject widget : widgets) {
+      Assertion.assertTrue(widget.isLoadedOnMercury(), MercuryMessages.INVISIBLE_MSG);
+    }
+  }
+
+  @Test(groups = "MercuryWidgetsCombinedTest_002")
+  @Execute(onWikia = "mercuryautomationtesting")
+  public void MercuryWidgetsCombinedTest_002_isLoadedOnFirstVisitFromDifferentArticle() {
+    String content = "";
+    for (WidgetPageObject widget : widgets) {
+      content += widget.getTag();
+    }
+
+    ArticlePageObject article = new ArticlePageObject(driver);
+    BasePageObject base = new BasePageObject(driver);
+    base.openMercuryArticleByNameWithCbAndNoAds(wikiURL, MercurySubpages.MAIN_PAGE);
+    article.open(ARTICLE_NAME);
+
+    for (WidgetPageObject widget : widgets) {
+      Assertion.assertTrue(widget.isLoadedOnMercury(), MercuryMessages.INVISIBLE_MSG);
+    }
+  }
+
+  @Test(groups = "MercuryWidgetsCombinedTest_003")
+  @Execute(onWikia = "mercuryautomationtesting")
+  public void MercuryWidgetsCombinedTest_003_isLoadedOnSecondVisitFromDifferentArticle() {
+    String content = "";
+    for (WidgetPageObject widget : widgets) {
+      content += widget.getTag();
+    }
+
+    ArticlePageObject article = new ArticlePageObject(driver);
+    article.open(ARTICLE_NAME);
+    BasePageObject base = new BasePageObject(driver);
+    base.openMercuryArticleByNameWithCbAndNoAds(wikiURL, MAPS_ARTICLE_NAME);
+    article.open(ARTICLE_NAME);
```

Comment:
those 5 lines can be replaced by 3
 `new ArticlePageObject(driver).openMercuryArticleByNameWithCbAndNoAds(ARTICLE_NAME)`
 `new ArticlePageObject(driver).openMercuryArticleByNameWithCbAndNoAds(MAPS_ARTICLE_NAME)`
 `new ArticlePageObject(driver).openMercuryArticleByNameWithCbAndNoAds(ARTICLE_NAME)`